docs/how_to_guides/custom_validators.md

```diff
@@ -0,0 +1,303 @@
+# Custom Validators
+
+If you need to perform a validation that is not currently supported by the hub, you can create your own custom validators.
+
+## As A Function
+A custom validator can be as simple as a single function if you do not require additional arguments:
+
+```py
+from typing import Dict
+from guardrails.validators import (
+    FailResult,
+    PassResult,
+    register_validator,
+    ValidationResult,
+)
+
+@register_validator(name="toxic-words", data_type="string")
+def toxicWords(value: str, metadata: Dict) -> ValidationResult:
+    if value.startswith("a"):
+        return PassResult()
+
+    for word in ["butt", "poop", "booger"]:
+        if word in value:
+            mentioned_words.append(word)
+
+    if len(mentioned_words) > 0:
+        return FailResult(
+            error_message=f"Mention toxic words: {', '.join(mentioned_words)}",
+        )
+    else:
+        return PassResult()
```

Comment:
is there a simpler version of this that doesn't require passresult or failresult?

Reply:
The return type from a validator has to be either a `PassResult` or `FailResult`.  The only way to simplify it would be for it to always pass or always fail which isn't a good example imo.